packages/gatsby/src/internal-plugins/query-runner/query-queue.js

```diff
@@ -1,106 +1,71 @@
 const Queue = require(`better-queue`)
 
 const queryRunner = require(`./query-runner`)
-const { store, emitter } = require(`../../redux`)
+const { store } = require(`../../redux`)
 const { boundActionCreators } = require(`../../redux/actions`)
 const websocketManager = require(`../../utils/websocket-manager`)
 const FastMemoryStore = require(`./better-queue-custom-store`)
 
-const processing = new Set()
-const waiting = new Map()
-
-const queueOptions = {
-  concurrent: 4,
-  // Merge duplicate jobs.
-  merge: (oldTask, newTask, cb) => {
-    cb(null, newTask)
-  },
-  priority: (job, cb) => {
-    const activePaths = Array.from(websocketManager.activePaths.values())
-    if (job.id && activePaths.includes(job.id)) {
-      cb(null, 10)
-    } else {
-      cb(null, 1)
-    }
-  },
-  // Filter out new query jobs if that query is already running.  When the
-  // query finshes, it checks the waiting map and pushes another job to
-  // make sure all the user changes are captured.
-  filter: (job, cb) => {
-    if (processing.has(job.id)) {
-      waiting.set(job.id, job)
-      cb(`already running`)
-    } else {
-      cb(null, job)
-    }
-  },
-  store: FastMemoryStore(),
-}
-
-// During builds we don't need all the filtering, etc. so we
-// remove them to speed up queries
-if (process.env.gatsby_executing_command === `build`) {
-  delete queueOptions.filter
-  delete queueOptions.priority
-  delete queueOptions.merge
+const makeOptions = ({ processing, waiting }) => {
+  return {
+    concurrent: 4,
+    // Merge duplicate jobs.
+    merge: (oldTask, newTask, cb) => {
+      cb(null, newTask)
+    },
+    priority: (job, cb) => {
+      const activePaths = Array.from(websocketManager.activePaths.values())
+      if (job.id && activePaths.includes(job.id)) {
+        cb(null, 10)
+      } else {
+        cb(null, 1)
+      }
+    },
+    // Filter out new query jobs if that query is already running.  When the
+    // query finshes, it checks the waiting map and pushes another job to
+    // make sure all the user changes are captured.
+    filter: (job, cb) => {
+      if (processing.has(job.id)) {
+        waiting.set(job.id, job)
+        cb(`already running`)
+      } else {
+        cb(null, job)
+      }
+    },
+    store: FastMemoryStore(),
+  }
 }
```

Comment:
I think we want to keep those - those add extra overhead which is only needed in watch mode - but for production builds we don't really benefit from those

Reply:
Oh wow, that was accidental. I didn't mean to delete that part. Great catch

Reply:
Resolved in https://github.com/gatsbyjs/gatsby/pull/12891/commits/60d1241014458bb8e3cb1dc517a7ba323201aad2. In my future branch, the websocket stuff is passed in by gatsby develop, so cleans up this file further.